src/toil/common.py

```diff
@@ -760,6 +765,16 @@ def getWorkflowDir(workflowID, configWorkDir=None):
         :rtype: str
         """
         workDir = configWorkDir or os.getenv('TOIL_WORKDIR') or tempfile.gettempdir()
+        if workDir == '.':
```

Comment:
Scratch that. Remove altogether. But in executor.py, in runJob, ensure that TOIL_WORKDIR is set to os.getcwd()  if its not set already. I would even go so far as to raise an exception here if workDir is not absolute. Maybe that's too much.